### ISSUE TITLE
Colour of smoke test block in build pipeline does not reflect failed status correctly.

### ISSUE THREAD

**Issue by sxa** [adoptium/temurin-build#3583]
**What are you trying to do?** Run a build and see the status reflected in the blocks in the pipeline overview

**Expected behaviour:** If the smoke test failes causing an overall fail status in the pipeline, then the corresponding square in the pipeline should show  failed status

**Observed behaviour:**
![image](https://github.com/adoptium/temurin-build/assets/6487691/6c9e1fdf-9e1f-47be-adde-d8a289b509d3)
This screenshot is despite the failure of [riscv64-temurin_SmokeTests#14](https://ci.adoptium.net/job/build-scripts/job/jobs/job/evaluation/job/jobs/job/jdk/job/jdk-evaluation-linux-riscv64-temurin_SmokeTests/14) triggered by [jdk-evaluation-linux-riscv64-temurin#47](https://ci.adoptium.net/job/build-scripts/job/jobs/job/evaluation/job/jobs/job/jdk/job/jdk-evaluation-linux-riscv64-temurin/47/console) (it's the overview for jdk-evaluation-linux-riscv64-temurin where the screen shot above was visible)

**Any other comments:**

**Comment by sophia-guo** [comment]
This is interesting. I took a look and seems stageResult typically reflects the immediate outcome of each stage based on the stage execution itself, independent of the build status. That is saying this is the expected behaviour.  I did check a few other pipeline jobs and it's the same behaviour. For example https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk17u/job/jdk17u-windows-x64-temurin/  build 421 failed due to the stage sign. The stage sign itself shows suceeds.
<img width="1252" alt="Screenshot 2023-12-18 at 10 36 17 AM" src="https://github.com/adoptium/temurin-build/assets/18030625/c65f9653-fcd1-4968-8f16-b8d962e2597a">

**Comment by sxa** [comment]
Noting that @sophia-guo suggested on a call this week thgat this was due to the fact it was a separate job being kicked off and that was causing the stage to not show the correct status by default.
There was also mention of the fact that TRSS can show these statuses more accurately, but I feel we should still try and make these jobs show accurate information (particularly if a job fails completely) and it also makes it easier to look at the output of a job not run from the regular higher level pipelines.

**Comment by sxa** [comment]
> which I expect will show the same.

Yep - while the smoke test now passes, sanity.system shows the problem: https://ci.adoptium.net/job/Test_openjdk21_hs_sanity.system_riscv64_linux/38/ failed
```
10:59:40  Build [Test_openjdk21_hs_sanity.system_riscv64_linux #38](https://ci.adoptium.net/job/Test_openjdk21_hs_sanity.system_riscv64_linux/38/) completed: FAILURE
```
![image](https://github.com/adoptium/temurin-build/assets/6487691/5c81bfbb-df07-45f2-832c-a0193fcc6080)
